mapping/sustainability.yaml

```
@@ -57,15 +57,120 @@
     module: Economic and fiscal
     indicator: Economic viability
     disclosure format: Name the funding organization(s)/sources of funding for Preparation, Implementation and Operation. If the information is not available for any of the stages, select \['funding/budget source not specified'\] for the respective stage where the nature of the funding/budget source could not be identified.
-    mapping: ''
-    example: ''
+    mapping: |-
+        Project level:
+
+        1. Get the `BudgetBreakdowns` object in the `budget.budgetBreakdowns` array that represents the budget breakdown by stage. If none exists yet:
+
+        - Add a `BudgetBreakdowns` object to the `budget.budgetBreakdowns` array.
+
+        2. Set it's `.id` incrementally and it's `.description` to "Breakdown by stage".
+
+        3. Get the `BudgetBreakdown` object in the budget breakdowns object for each of the 3 stages. If none exists yet:
+
+        - Add 3 `BudgetBreakdown` objects to the `budget.budgetBreakdowns.budgetBreakdown` array. For each object:
+        - Set the `.id` incrementally.
+        - Set the `.description` to the stage name - "Preparation", "Implementation" or "Operation". If the funding source is not known for a stage note append "funding/budget source not specified" to the `.description`.
+
+        4. For each stage, get the `Organization` object in `.parties` that represents the party providing the funding for the stage named in `.description`. If none exists yet, [add an organization](../common.md#add-an-organization) and add 'funder' to its `.roles`.
+
+        - Set the budget breakdown's `.sourceParty` to the `.id` and `.name` of the organization.
+    example: |-
+        {
+          "budget": {
+            "budgetBreakdowns": [
+              {
+                "id": "1",
+                "description": "Breakdown by stage",
+                "budgetBreakdown": [
+                  {
+                    "id": "1",
+                    "description": "Preparation",
+                    "sourceParty": {
+                      "id": "1",
+                      "name": "Agency for Agricultural Development of Morocco"
+                    }
+                  },
+                  {
+                    "id": "2",
+                    "description": "Implementation",
+                    "sourceParty": {
+                      "id": "1",
+                      "name": "Agency for Agricultural Development of Morocco"
+                    }
+                  },
+                  {
+                    "id": "3",
+                    "description": "Operation funding/budget source not specified"
+                  }
```

In keeping with the OC4IDS approach of omitting missing data rather than setting fields to "n/a" or similar.

```suggestion
                  }
```